Shopify/cli
# Updated command documentation

### WHY are these changes introduced?
As part of the project to centralize CLI docs, this PR includes detailed documentation for each command.

In the future the CLI repo will be the source of truth for the CLI documentation.

Contributes to https://github.com/Shopify/develop-app-management/issues/1637



### WHAT is this pull request doing?
- Add detailed documentation for each command under the `description` property.
- Add a short 1-phrase description in the `summary`property for each command.
- The information added in this PR is the same you can find in shopify.dev



### How to test your changes?
If you run `--help` on any command you'll see a detailed description:

`pnpm shopify app dev --help`



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
